MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java

```diff
@@ -435,6 +441,14 @@ public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float ve
       // calculate animation time based on displacement
       long animationTime = (long) (velocityXY / 7 / tiltFactor + MapboxConstants.ANIMATION_DURATION_FLING_BASE);
 
+      if (!uiSettings.isHorizontalScrollGesturesEnabled()) {
+        offsetX = 0.0;
+        if (offsetY < MapboxConstants.Y_TRANSLATION_THRESHOLD_IGNORE_FLING) {
```

Comment:
added this option to ignore flings if they happen mostly horizontally, the value is not really backed up by any scientific measurement.

Reply:
thinking about this today, I should calculate the angle of the fling against the horizontal axis and determine a threshold for that given angle to not execute a fling